scripts/helpers/civitai_utils.py

```diff
@@ -0,0 +1,191 @@
+    :return: sha_256 hash as str
+    """
+    json_file = os.path.splitext(file_path)[0] + ".json"
+
+    if os.path.exists(json_file):
+        try:
+            with open(json_file, 'r', encoding="utf-8") as f:
+                data = json.load(f)
+
+            if 'sha256' in data and data['sha256']:
+                hash_value = data['sha256']
+                return hash_value
+        except Exception as e:
+            out(f"Failed to open {json_file}; {e}")
+
+    def read_chunks(file, size=io.DEFAULT_BUFFER_SIZE):
+        while True:
+            chunk = file.read(size)
+            if not chunk:
+                break
+            yield chunk
+
+    blocksize = 1 << 20
+    h = hashlib.sha256()
+    length = 0
+    with open(os.path.realpath(file_path), 'rb') as f:
+        for block in read_chunks(f, size=blocksize):
+            length += len(block)
+            h.update(block)
+
+    hash_value = h.hexdigest()
+
+    if os.path.exists(json_file):
+        try:
+            with open(json_file, 'r', encoding="utf-8") as f:
+                data = json.load(f)
+
+            if 'sha256' in data and data['sha256']:
+                data['sha256'] = hash_value
+
+            with open(json_file, 'w', encoding="utf-8") as f:
+                json.dump(data, f, indent=4)
+        except Exception as e:
+            out(f"Failed to open {json_file}: {e}")
+    else:
+        data = {'sha256': hash_value}
+        with open(json_file, 'w', encoding="utf-8") as f:
+            json.dump(data, f, indent=4)
+
+    return hash_value
+
+
+def query_model_info(file_path):
+    model_hash = gen_sha256(file_path)
+    model_info = request_civit_api(f"{api_urls['model_hash']}{model_hash}")
+    try:
+        model_id = model_info["modelId"]
+
+    except KeyError:  # Failed query
+        out(f"Failed to retrieve model info for {file_path}")
+        return file_path, {}
+
+    model_info = request_civit_api(f"{api_urls['model_id']}{model_id}")
+    try:
+        return file_path, model_info
+    except KeyError:
+        out(f"Failed to retrieve tags for {file_path}")
+        return file_path, {}
+
+
+def model_info_query(file_paths):
+    if len(file_paths) <= 0:
+        return file_paths
+
+    paths_and_info = []
+    for file in file_paths:
+        paths_and_info.append(query_model_info(file))
+
+    return paths_and_info
+
+
+def add_civitai_tags():
+    for network in networks:
+        #path = make_network_path(network)
+        path = model_dirs[network]
+
+        if not os.path.exists(path):
+            out(f"No folder for {network} found in models directory")
+            continue
+
+        files = [file for file in os.listdir(path) if os.path.splitext(file)[1].lstrip(".") in networks[network]]
+        file_paths_to_write_info = [os.path.join(path, file) for file in files if not os.path.exists(
+            os.path.join(path, f"{os.path.splitext(os.path.basename(file))[0]}.civitai.info"))]
```

Comment:
I'm not quite I understand what's going on here. Should the file not be queried if a corresponding `.civitai.info` file is found? If anything, we can use the hash stored within the file to skip the computationally heavy sha256 hashing step.